themes/ddbasic/sass/components/header.scss

```diff
@@ -0,0 +1,596 @@
+    top: 0;
+    left: 0;
+    width: 100%;
+    background-color: $grey-light;
+    box-shadow: $box-shadow;
+    
+    // Tablet
+    @include media($tablet){
+      box-shadow: none;
+    }
+    
+    // Toolbar
+    .toolbar & {
+      top: $toolbar-height;
+    }
+    .topbar-inner {
+      @include wrapper;
+      @include transition(height $speed $ease);
+      position: relative;
+      height: 64px;
+      
+      // Tablet
+      @include media($tablet) {
+        width: 100%;
+      }
+      
+      .logo {
+        @include span-columns(3);
+        height: 44px;
+        margin-top: 10px;
+        .site-name-container {
+          height: 44px;
+          .site-name {
+            @include font('header-small');
+            line-height: 44px;
+            color: $charcoal;
+          }
+          &.logo-container {
+            .site-name {
+              display: none; 
+            }
+            .vertical-helper {
+              display: inline-block;
+              height: 100%;
+              vertical-align: middle;
+            }
+            img {
+              display: inline-block;
+              //float: left;
+              vertical-align: middle;
+              max-height: 100%;
+              max-width: 95%;
+              width: auto;
+              height: auto;
+            }
+          }
+        }
+        
+        // Tablet
+        @include media($tablet) {
+          position: relative;
+          width: 100%;
+          height: 64px;
+          margin-top: 0;
+          padding: 10px 5%;
+          box-shadow: $box-shadow;
+          background-color: $grey-light;
+          z-index: 5;
+          .site-name-container {
+            width: 100%;
+          }
+        }
+        // Tablet
+        @include media($mobile) {
+          .site-name-container {
+            &.logo-container {
+              .site-name {
+                display: block; 
+              }
+              .vertical-helper,
+              img {
+                display: none;
+              }
+            }
+          }
+        }
+      }
+      .topbar-menu {
+        
+        // Tablet
+        @include media($tablet) {
+          width: 100%;
+          position: relative;
+          z-index: 4;
+          float: left;
+          padding: 10px 5%;
+          background-color: $color-primary;
+          box-shadow: $box-shadow;
+        }
+        
+      }
+      .pane-user-name {
+        float: right;
+        width: getColumn(4);
+        margin-right: getGutter(12)/2;
+        
+        // Tablet
+        @include media($tablet){
+          display: none;
+        }
+      }
+    }
+  }
+  // Header wrapper
+  .header-wrapper {
+    position: fixed;
+    z-index: $z-header - 1;
+    top: $topbar-height;
+    width: 100%;
+    // Tablet
+    @include media($tablet) {
+      z-index: $z-header - 1;
+    }
+    // Search form extended
+    .search-form-extended & {
+      position: static;
+    }
+    .header-inner {
```

Comment:
A lot of stuff is going on in the header-inner regarding positioning and sizing across multiple breakpoints and multiple transitions. This could do with a cleanup, maybe som clear sectioning or further commenting. It all seems very messy to an outsider. Why -102%?